Redraw schematic to match functional diagram from model (no electrical / PCB changes)

pddenhar opened the pull request:
I found this project on the vintagehondatwins forums; thanks so much for sharing this awesome reverse engineering work you did with the community. This knowledge goes a long way towards keeping these old bikes around. 

In that vein, I thought it would be a fun challenge to redraw the Rev C schematic to more closely match the functional diagram in the service manual. I'm hoping by doing this the principles of how the CDI unit functions will be a bit more accessible to people who want to understand it. 
![image](https://github.com/user-attachments/assets/7027e372-beb4-4c7b-a697-d78b7eea03cc)

I was very careful to do this without changing any of the actual structure of the circuit. I took before and after screenshots of the PCB (without changing any traces) and checked to make sure that it passed DRC checks after my revisions to the drawings. 

Schematic and PCB before:
![Screenshot 2025-03-28 192802](https://github.com/user-attachments/assets/618fed4d-4697-42f3-a8f4-baaf4cbbc346)
![Screenshot 2025-03-28 192747](https://github.com/user-attachments/assets/6409a404-80b7-4dd5-821f-124ff12842af)

Schematic and PCB after:
![Screenshot 2025-03-28 192550](https://github.com/user-attachments/assets/5e356d2a-1358-4634-a089-ae7bf9d2ee14)
![Screenshot 2025-03-28 192526](https://github.com/user-attachments/assets/db4268a8-2c8e-4515-8740-fdda6ef92d67)

PDF of my revised schematic: 
[CM400T_CDI_Revision_C.pdf](https://github.com/user-attachments/files/19514854/CM400T_CDI_Revision_C.pdf)


As you can see, I also named most of the important nets to make the functionality more apparent at a glance. While I was in there I also created a custom footprint file for the 35mm capacitor so that the PCB editor DRC would stop complaining about it 😄 

Thanks again for sharing this great project and I hope this is useful!

pddenhar committed:
Redraw schematic to match structure of diagram from manual. No electrical / functional changes

pddenhar committed:
Add footprint file for 35mm capacitor to fix DRC violation

maraakate commented:
Hi, I appreciate your work.  My only question (I have not tried to use the PR) is that in the second picture of the PCB layout it now shows P1, P2, etc. instead of the colour codes.  If this is not a view issue, but rather you changed the labels can you please change it back to colours?  One, for easier assembly and two, the original PCBs actually show the colour codes, but they are in Japanese.

Otherwise, great work and will help those to study and learn from it!

pdenhartog committed:
Add ColorCode field to wire hole symbols so that code can be displayed on silkscreen instead of reference designator

pddenhar commented:
Great catch, those silkscreen labels are auto-populated from the reference designator in the schematic and get carried over any time you run "Update PCB from Schematic" (which I was doing constantly to make sure no PCB tracks or component connections were changing).

Since it's P5 here:
![image](https://github.com/user-attachments/assets/619bfda8-ba2c-4930-9423-dd1513065764)

It gets reverted to P5 in the PCB editor:
![image](https://github.com/user-attachments/assets/2bb0380a-e05d-4049-acd7-0a3ebc0851a4)

I added a symbol field called ColorCode in the schematic and set it to display on the silkscreen of the board instead of the symbol reference designators: 
![image](https://github.com/user-attachments/assets/d271abf1-4bbb-496d-b0c2-27ce99489fb7)
![image](https://github.com/user-attachments/assets/12462c85-9b0f-4906-8be0-f9dffe6b57f9)


This way these labels will be persistent and won't revert any time the schematic is changed.

Corrected board from latest commit:
![image](https://github.com/user-attachments/assets/9b91cca1-c54b-4d07-94c5-5fed74bf5973)

pddenhar commented:
Something I **didn't** change but wanted to mention:

In the original version of the schematic (without any of my changes), C5 has a 12.5mm long footprint selected:
![image](https://github.com/user-attachments/assets/0da2b352-51c1-476c-9365-2e75a1f24512)

But the footprint on the PCB is a 15mm long axial capacitor:
![image](https://github.com/user-attachments/assets/3f1cf5fe-fcec-4754-b0be-9de425fc03f3)

I left it as a **15mm** footprint since that's the way it was on the PCB already. Just wanted to mention it in case something different was intended.

maraakate commented:
What is on the PCB now is what the actual size is because I've made over 50 of these with this layout.  So, I suppose this means 15mm is the correct size and needs changed?

pddenhar commented:
> What is on the PCB now is what the actual size is because I've made over 50 of these with this layout. So, I suppose this means 15mm is the correct size and needs changed?

Sounds good. This pull request will fix this as well then. I updated the schematic symbol to match what was in the PCB since having a mismatch caused DRC errors. 

Since you've been manufacturing them already with the 15mm footprint that was in the PCB, it sounds like it's correct that way! 
![image](https://github.com/user-attachments/assets/ae22634c-365b-4d7b-a185-97fe3a98aa77)

maraakate commented:
Agreed.  Sounds good to me.  I'll merge it.  Thanks a lot for your enhancements.  I'm just a hacker so I had figure out KiCAD on my own.
